src/torrent/utils/log.cc

```diff
@@ -152,17 +151,11 @@ log_group::internal_print(const HashString* hash, const char* subsystem, const v
 
   pthread_mutex_lock(&log_mutex);
 
-  std::for_each(m_first, m_last, std::bind(&log_slot::operator(),
-                                           std::placeholders::_1,
-                                           (const char*)buffer,
-                                           std::distance(buffer, first),
-                                           std::distance(log_groups.begin(), this)));
+  std::for_each(m_first, m_last, [this, begin = static_cast<const char*>(first), data = static_cast<const char*>(buffer)](auto log) {
+    log(data, std::distance(data, begin), std::distance(log_groups.begin(), this));
```

Comment:
Calculate the `std::distance` outside the for loop.

Reply:
Done.